common/src/main/java/com/wynntils/overlays/StatusEffectsOverlay.java

```diff
@@ -108,4 +135,59 @@ private void updateTextRenderSetting() {
     protected TextRenderSetting getTextRenderSetting() {
         return textRenderSetting;
     }
+
+    private Stream<RenderedStatusEffect> stackEffects(List<StatusEffect> effects) {
+        Map<String, RenderedStatusEffect> effectsToRender = new LinkedHashMap<>();
+
+        for (StatusEffect effect : effects) {
+            RenderedStatusEffect entry =
+                    effectsToRender.get(effect.asString().getString());
+            if (entry == null) {
+                entry = new RenderedStatusEffect(effect);
+                effectsToRender.put(effect.asString().getString(), entry);
+            }
+            entry.count += 1;
+        }
+        return effectsToRender.values().stream();
+    }
+
+    private static final class RenderedStatusEffect implements Comparable<RenderedStatusEffect> {
+        private int count = 0;
+        private final StatusEffect effect;
+
+        private RenderedStatusEffect(StyledText name, StyledText modifier, StyledText displayedTime, StyledText prefix) {
+            this.effect = new StatusEffect(name, modifier, displayedTime, prefix);
+        }
+
+        private RenderedStatusEffect(StatusEffect effect) {
+            this.effect = effect;
+        }
+
+        private StyledText getTextToRender() {
+            return this.count > 1
+                    ? StyledText.fromString(this.count + "x ").append(this.effect.asString())
+                    : this.effect.asString();
+        }
+
+        public int getCount() {
+            return this.count;
+        }
+
+        public void setCount(int c) {
+            this.count = c;
+        }
+
+        public StatusEffect getEffect() {
+            return this.effect;
+        }
+
+        @Override
+        public int compareTo(@NotNull RenderedStatusEffect effect) {
+            return ComparisonChain.start()
+                    .compare(this.effect.getPrefix().getString(), effect.effect.getPrefix().getString())
+                    .compare(this.effect.getName().getString(), effect.effect.getName().getString())
+                    .compare(this.effect.getModifier().getString(), effect.effect.getModifier().getString())
+                    .result();
+        }
```

Comment:
I might not have been clear the first time around, but please implement this in the real `StatusEffect` class.

Also, we disallow using `@NotNull`, so please remove it.

Reply:
Alright

The `@NotNull` was inserted there by IntellIJ, I'll move it.